pkg/defaults/catsrcHelpers.go

```diff
@@ -64,7 +64,9 @@ func processCatsrc(client wrapper.Client, def olm.CatalogSource, disable bool) e
 	}
 
 	if disable {
-		err = ensureCatsrcAbsent(client, def, cluster)
+		if cluster.Annotations[defaultCatsrcAnnotationKey] == defaultCatsrcAnnotationValue {
```

Comment:
This probably requires that the default `CatalogSource` manifests contain the annotation: https://github.com/operator-framework/operator-marketplace/tree/master/defaults

Reply:
The annotation is added at runtime instead of requiring it to be present on the canonical default manifests.